### PR TITLE
Interaction net updates

### PR DESCRIPTION
**Pull request by cwgoes**
Closes https://github.com/cryptiumlabs/juvix/issues/158

### CODE REVIEW

**Comment by cwgoes** [comment]
@mariari I'll add some more testcases, but want to quickly check if this is the right approach. We should probably have the interpreter mirror the LLVM backend as closely as possible (ideally we can even share a lot of the intermediate construction code).

File: src/Juvix/Interpreter/InteractionNet/Nets/Default.hs [hunk 1]
```diff
@@ -563,9 +596,25 @@ curry2 (nodeF, numNode) (p, numPrim) = do
...
+  incGraphSizeStep (- 1)
+  curr ← newNode (Auxiliary1 $ PrimCurried1 $ nodeF p)
+  let currNode = RELAuxiliary1
+        { node = curr,
+          primary = ReLink numNode Aux2,
+          auxiliary1 = ReLink numNode Aux1
+        }
+  linkAll currNode
+  deleteRewire [numNode, numPrim] [curr]
+
```
Comment (mariari) [inline src/Juvix/Interpreter/InteractionNet/Nets/Default.hs]: What is the difference between this and curry2?

File: src/Juvix/Interpreter/InteractionNet/Nets/Default.hs [hunk 2]
```diff
@@ -587,8 +636,28 @@ curry1 (nodeF, numNode) (p, numPrim) =
...
+      incGraphSizeStep (- 1)
+      curr ← newNode (Primar node)
+      let currNode = RELAuxiliary0
+            { node = curr,
+              primary = ReLink numNode Aux1
+            }
+      linkAll currNode
+      deleteRewire [numNode, numPrim] [curr]
+      pure $ True
+
```
Comment (mariari) [inline src/Juvix/Interpreter/InteractionNet/Nets/Default.hs]: Same for this vs curry1

File: src/Juvix/Interpreter/InteractionNet/Nets/Default.hs (hunk 1 above)
Comment (cwgoes) [reply src/Juvix/Interpreter/InteractionNet/Nets/Default.hs]: They should be de-duplicated, we'll get rid of `Primitive`.

File: src/Juvix/Interpreter/InteractionNet/Nets/Default.hs (hunk 2 above)
Comment (cwgoes) [reply src/Juvix/Interpreter/InteractionNet/Nets/Default.hs]: Ditto on the answer.

**Comment by cwgoes** [comment]
Follow-up issue: https://github.com/cryptiumlabs/juvix/issues/184

File: src/Juvix/Interpreter/InteractionNet/Nets/Default.hs (hunk 1 above)
Comment (cwgoes) [reply src/Juvix/Interpreter/InteractionNet/Nets/Default.hs]: Noted in https://github.com/cryptiumlabs/juvix/issues/184.

File: src/Juvix/Interpreter/InteractionNet/Nets/Default.hs (hunk 2 above)
Comment (cwgoes) [reply src/Juvix/Interpreter/InteractionNet/Nets/Default.hs]: Noted in https://github.com/cryptiumlabs/juvix/issues/184.

**Pull request merged**